### PR TITLE
Final Review

### PR DESCRIPTION
**Pull request by PowerLevel9000**
# Final Review For Coder Pay

## Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
  
 - Documentation 
 - Testing

## Happy coding

CoolImGIF (2)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,233 @@
...
+
+If you like this project please give it star or buy a coffee for us
+
+[![BuyMeACoffee](https://img.shields.io/badge/Buy%20Me%20a%20Coffee-ffdd00?style=for-the-badge&logo=buy-me-a-coffee&logoColor=black)](https://buymeacoffee.com/adi8090808E)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
```
Comment (rotimiazeez) [inline README.md:217]: Please ensure your"Acknowledgements" section gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

Comment (rotimiazeez) [reply README.md:217]: - Please ensure you deploys the application using `Render` or any other deployment platform.
- Also ensure project is deployed properly, the home and detail page works as expected and a link to the deployed project is present in your README.

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @PowerLevel9000 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- Good project presentation video 📹
- Good job adding the splash screen as required 👌🏽
-  users  can `Login` and `sign-up` on the app 💯
- The `home` page and `transaction` are functioning as required 👏🏾
- Good app UI design 🖼
-Tests are implemented and are passing ✔️
- No Linter error ✔️ 
- Good PR description 👍🏽
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (PowerLevel9000) [reply README.md:217]: hi @rotimiazeez 

I already have one deployment running on the render with data base and it is asking for payment to deploy another can you please provide me helping hand regarding this

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PowerLevel9000** [comment]
# Thank you 💪💪💪🙏🙏🙏

ThankYouSoMuchPrabhasGIF


Dear @rotimiazeez @ichala ,

I wanted to express my gratitude for your willingness to take the time to review my code on GitHub. Your feedback and suggestions have been instrumental in improving the quality of my work, and I appreciate your insights and expertise in this area.

It's inspiring to have someone like you, who is knowledgeable and experienced, provide valuable feedback that helps me enhance my coding skills. I truly appreciate the time and effort you put into reviewing my code, and I'm grateful for your constructive criticism.

If there are any areas that I need to work on, please don't hesitate to let me know. I'm open to hearing any suggestions or recommendations you may have. Your feedback is essential to my growth as a developer, and I value it tremendously.

Thank you again for your time and for sharing your expertise with me. I'm looking forward to applying the insights gained from your review to my future projects.

Sincerely,
@PowerLevel9000

**Pull request merged**